### PR TITLE
7 ready for review

### PR DESCRIPTION
**Pull request by LKFULLER1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -175,21 +175,67 @@ describe('6. GET /api/reviews/:review_id/comments', () => {
 
 describe('7. POST /api/reviews/:review_id/comments', () => {
   test('status:201, responds with comment newly added to the database', () => {
+    const REVIEW_ID = 2;
     const newComment = {
       author: 'mallionaire',
       body: 'rubbish mate, grumble!',
+      review_id: REVIEW_ID
```
Comment (mezzDavies) [inline __tests__/app.test.js]: The ticket states the only things on the request body are the author and body properties. So we don't/shouldn't need to include the review ID here - it is in the URL as a parameter (and can be accessed by your controller on req.params).

File: controllers/games.js [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
+  })
     .catch((err) => {
       next(err)
     })
-  }
\ No newline at end of file
+}
+
+exports.postCommentByReviewId = (req, res, next) => {
+  insertCommentByReviewId(req.body).then((comment) => {
```
Comment (mezzDavies) [inline controllers/games.js]: The id needs to be taken from req.params not the request body. You have doen this in your getCommentsByReviewId controller above. You'll just need to pass more arguments to your insertCommentByReviewId model when you have got hold of it and the request.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -175,21 +175,67 @@ describe('6. GET /api/reviews/:review_id/comments', () => {
...
     };
     return request(app)
-      .post('/api/reviews/2/comments')
+      .post(`/api/reviews/${REVIEW_ID}/comments`)
```
Comment (mezzDavies) [inline __tests__/app.test.js]: We can just pass in the id in to the url - no variable needed.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -175,21 +175,67 @@ describe('6. GET /api/reviews/:review_id/comments', () => {
...
+      review_id: REVIEW_ID
+    };
+    return request(app)
+      .post(`/api/reviews/${REVIEW_ID}/comments`)
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('invalid query!');
+      })
+      .then(() => {
```
Comment (mezzDavies) [inline __tests__/app.test.js]: Could this be split into two separate tests?

File: __tests__/app.test.js [hunk 5]
```diff
@@ -175,21 +175,67 @@ describe('6. GET /api/reviews/:review_id/comments', () => {
...
-        expect(body.comment).toEqual({
-          comment_id: 6 ,
+        expect(body.comment).toMatchObject({
+          comment_id: 7,
           ...newComment,
         });
       });
   });
+
+  test('status:404, responds with a 404 error when the review_id does not exist', () => {
```
Comment (mezzDavies) [inline __tests__/app.test.js:194]: This test will need amending to accommodate for the ID being on req.params

**Review by mezzDavies — CHANGES_REQUESTED** [review]
I think you aren't quite on top of how this endpoint works regarding the ID in the URL. I've left some comments that should point you in the right direction 😄 . This needs amending for all the test/code of the endpoint thus far. 

Once that's done there a few more things to consider testing and accounting for too eg passing a valid but non-existent id eg 99999999999,  if the username doesn't exist. Please consider the error codes for these possibilities too. 👍

**Review by MKD01 — APPROVED** [review]
Looks good, im happy for this to be merged

**Pull request merged**